src/TreeRegression.cpp

```diff
@@ -44,13 +44,46 @@ void TreeRegression::allocateMemory() {
 double TreeRegression::estimate(size_t nodeID) {
 
   // Mean of responses of samples in node
-  double sum_responses_in_node = 0;
+  double sum_responses_in_node = sumNodeResponse(nodeID);
+
   size_t num_samples_in_node = end_pos[nodeID] - start_pos[nodeID];
-  for (size_t pos = start_pos[nodeID]; pos < end_pos[nodeID]; ++pos) {
-    size_t sampleID = sampleIDs[pos];
-    sum_responses_in_node += data->get_y(sampleID, 0);
+  if (splitrule == POISSON && sum_responses_in_node == 0.) {
+    // Poisson is not allowed to predict 0.
+    // We use a weighted average of parent and child mean values,
+    // see vignette "Introduction to Rpart" Chapter 8.2 and
+    // https://ssrn.com/abstract=2870308 Chapter 6.1.3
+    
+    // Search for parent's nodeID: loop over all nodeIDs
```

Comment:
If this turns out to be a performance bottleneck (might be in huge trees), we could save parent nodes while splitting.

Reply:
Just noticed this is used only when sum is 0, so I guess unlikely to be a bottleneck.

Reply:
If it turns out to be a bottleneck, your suggestion is a good idea. Until then and to keep this PR small, I would leave it as is.